scripts/pre-commit.sh

```diff
@@ -20,31 +20,83 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+
+function build-binaries() {
+  go build -o bin/controller-manager ./cmd/controller-manager
```

Comment:
(Action not necessary), we can use make commands now.
You can as well differ this to be done separately.